core/src/main/java/io/substrait/extension/SimpleExtension.java

```diff
@@ -729,7 +729,7 @@ public static ExtensionCollection load(List<String> resourcePaths) {
                   try (var stream = ExtensionCollection.class.getResourceAsStream(path)) {
                     return load(path, stream);
                   } catch (IOException e) {
-                    throw new RuntimeException(e);
+                    throw new IllegalStateException(e);
```

Comment:
Since IOException is the underlying exception, you might consider [UncheckedIOException](https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html) here.

Reply:
thanks, I changed the two occurrences.